error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -675,6 +675,34 @@ Flux<T> after(Mono<T> mono) {
     }
   }
 
+  /** Don't unnecessarily convert {@link Mono} to {@link Flux}. */
+  @SuppressWarnings("VoidMissingNullable")
+  static final class MonoFluxThen<T> {
+    @BeforeTemplate
+    Mono<Void> before(Mono<T> mono) {
+      return mono.flux().then();
+    }
+
+    @AfterTemplate
+    Mono<Void> after(Mono<T> mono) {
+      return mono.then();
+    }
+  }
+
+  /** Don't unnecessarily call {@link Mono#then()} on a {@code Mono<Void>} . */
+  @SuppressWarnings("VoidMissingNullable")
```

Comment:
Introduced this because I got into a fight with NullAway:
```
[NullAway] dereferenced expression mono is @Nullable
```
"mono" was nullable:
```
    @BeforeTemplate
    Mono<@Nullable Void> before(Mono<@Nullable Void> mono) {
      return mono.then();
    }
```
    
regardless, I wasn't really sure why we need `@Nullable` on the `Void` type in this context

Reply:
IIUC this is a bug in NullAway; indeed [their support for generics is limited](https://github.com/uber/NullAway/issues?q=is%3Aopen+generics). Let's add the `@Nullable` annotations (they are correct) but also keep the suppression (only on this class).